sdk/digitaltwins/digitaltwins/src/digitalTwinsClient.ts

```diff
@@ -0,0 +1,851 @@
+import { v4 } from "uuid";
+import { AzureDigitalTwinsAPI as GeneratedClient } from "./generated/src/azureDigitalTwinsAPI";
+import {
+  AzureDigitalTwinsAPIOptions,
+  DigitalTwinsGetByIdResponse,
+  DigitalTwinsAddOptionalParams,
+  DigitalTwinsAddResponse,
+  DigitalTwinsUpdateOptionalParams,
+  DigitalTwinsUpdateResponse,
+  DigitalTwinsDeleteMethodOptionalParams,
+  DigitalTwinsGetComponentResponse,
+  DigitalTwinsUpdateComponentResponse,
+  DigitalTwinsUpdateComponentOptionalParams,
+  DigitalTwinsAddRelationshipResponse,
+  DigitalTwinsAddRelationshipOptionalParams,
+  DigitalTwinsUpdateRelationshipOptionalParams,
+  DigitalTwinsUpdateRelationshipResponse,
+  DigitalTwinsDeleteRelationshipOptionalParams,
+  DigitalTwinsListIncomingRelationshipsResponse,
+  DigitalTwinsListRelationshipsOptionalParams,
+  DigitalTwinsListRelationshipsResponse,
+  DigitalTwinsSendTelemetryOptionalParams,
+  DigitalTwinsSendComponentTelemetryOptionalParams,
+  DigitalTwinModelsGetByIdResponse,
+  DigitalTwinModelsListOptionalParams,
+  DigitalTwinModelsListResponse,
+  DigitalTwinModelsListOptions,
+  DigitalTwinModelsAddResponse,
+  DigitalTwinModelsAddOptionalParams,
+  ModelData,
+  IncomingRelationship,
+  EventRoutesGetByIdResponse,
+  EventRoutesListOptionalParams,
+  EventRoutesListNextResponse,
+  EventRoute,
+  EventRoutesAddOptionalParams,
+  QueryQueryTwinsResponse,
+  QuerySpecification
+} from "./generated/src/models";
+
+/**
+ * Client for Azure IoT DigitalTwins API.
+ */
+export class DigitalTwinsClient {
+  /**
+   * @internal
+   * @ignore
+   * A reference to the auto-generated AzureDigitalTwinsAPI
+   */
+  private readonly client: GeneratedClient;
+
+  // private readonly client!: AzureDigitalTwinsAPI;
+
+  /**
+   * Creates an instance of AzureDigitalTwinsAPI.
+   *
+   * Example usage:
+   * ```ts
+   * const { DigitalTwinsClient, ServiceClientCredentials } = require("@azure/digitaltwins");
+   *
+   * const client = new DigitalTwinsClient(
+   *   "<endpoint>",
+   *   new DefaultAzureCredential();
+   * );
+   * ```
+   * @param {string} endpoint The endpoint of the service. If options parameter is given than this value will be ignored.
+   * @param {TokenCredential | ServiceClientCredentials} credential Used to authenticate requests to the service.
+   * @param {AzureDigitalTwinsAPIOptions} [options] Used to configure the service client.
+   */
+  constructor(
+    enpoint: string,
+    credential: TokenCredential | ServiceClientCredentials,
+    options?: AzureDigitalTwinsAPIOptions
+  ) {
+    if (!options) {
+      const options: AzureDigitalTwinsAPIOptions = {};
+      options.baseUri = enpoint;
+    }
+
+    this.client = new GeneratedClient(credential, options);
+  }
+
+  /**
+   * Get a digital twin
+   *
+   * @param digitalTwinId The Id of the digital twin.
+   * @returns The application/json digital twin and the http response.
+   */
+  public getDigitalTwin(digitalTwinId: string): Promise<DigitalTwinsGetByIdResponse> {
```

Comment:
There's more work here if you want to support tracing, e.g. you have to create an OpenTelemetry span and wrap around calling the underlying generated client. 

Take a look at how Text Analytics does this: https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/textanalytics/ai-text-analytics/src/textAnalyticsClient.ts#L241

Reply:
Yes, that work is ahead.

---
In reply to: [447259952](https://github.com/Azure/azure-sdk-for-js/pull/9787#discussion_r447259952) [](ancestors = 447259952)